civic_scraper/platforms/legistar/scraper.py

```diff
@@ -0,0 +1,80 @@
+import civic_scraper
+from civic_scraper import base
+from civic_scraper.base.asset import Asset, AssetCollection
+from civic_scraper.base.cache import Cache
+
+from legistar.events import LegistarEventsScraper
+
+from datetime import datetime
+from pathlib import Path
+from urllib.parse import urlparse, parse_qs
+
+# Scrape today's agendas and minutes from a Legistar site
+class LegistarSite(base.Site):
+
+    def __init__(self, base_url, municipality, cache=Cache(), parser_kls=None, timezone=None):
+        super().__init__(base_url, cache, parser_kls)
+        self.municipality = municipality
+        self.timezone = timezone
+
+    def create_asset(self, event, scraper):
+        # get date and time of event
+        meeting_datetime = " ".join((event['Meeting Date'], event['Meeting Time']))
+        meeting_date = scraper.toDate(meeting_datetime)
+        meeting_time = scraper.toTime(meeting_datetime)
+
+        # get event ID
+        if type(event['Meeting Details']) == dict:
+            url = event['Meeting Details']['url']
+            query_dict = parse_qs(urlparse(url).query)
+
+            meeting_id = 'legistar_{}_{}'.format(self.municipality, query_dict['ID'][0])
+        else:
+            # No meeting details, e.g., event is in future
+            url = None
+            meeting_id = None
+
+        # get event name
+        if type(event['Name']) == dict:
+            asset_name = event['Name']['label']
+            committee_name = event['Name']['label']
+        else:
+            asset_name = event['Name']
+            committee_name = event['Name']
+
+        e = {'url': url,
+             'asset_name': asset_name,
+             'committee_name': committee_name,
+             'place': event['Meeting Location'],
+             'state_or_province': self.municipality,
+             'asset_type': 'Agenda',
+             'meeting_date': meeting_date,
+             'meeting_time': meeting_time,
+             'meeting_id': meeting_id,
+             'scraped_by': f'civic-scraper_{civic_scraper.__version__}',
+             'content_type': 'txt',
+             'content_length': None,
+        }
+        return Asset(**e)
+
+    def scrape(self, download=True):
+        webscraper = LegistarEventsScraper(retry_attempts=3)
+
+        webscraper.BASE_URL = "https://canton.legistar.com/"
```

Comment:
are you using base_url anywhere? if not , then don't set it. if so, you almost certainly don't want it hard coded to cantont.

Reply:
ah, yes, thanks for pointing out Canton being hard-coded. I'll push up a change for that.

the BASE_URL is necessary in order to instantiate the webscraper. It's used in the [`LegistarScraper` base class](https://github.com/opencivicdata/python-legistar-scraper/blob/master/legistar/base.py#L221), which the `LegistarEventsScraper` class inherits from.

To this point, is there a better place to add this url to `webscraper.BASE_URL`? I did add a comment in the meantime explaining why it's there.